expcore/Mini_games.lua

```diff
@@ -100,73 +108,117 @@ function Mini_games.get_running_game()
     return started_game[1]
 end
 function Mini_games.start_game(name,parse_args)
-    local mini_game = Mini_games.mini_games[name]
-    if mini_game == nil then
-        return "This mini_game does not exsit"
-    end
-
-    if parse_args then
-        if  mini_game.options ~= #parse_args then
-            return "Wrong number of arguments"
+    game.print(parse_args)
+    game.print(vars.is_lobby )
+    if vars.is_lobby == true  or vars.is_lobby =="true" then
+        game.print('hi')
+        local player_names = {}
+        local server_object = global.servers[name]
+        local server = server_object[#server_object]
+        for _, player in ipairs(game.connected_players) do
+            player.connect_to_server{address=server,name=name}
+            player_names[#player_names+1 ] = player.name
         end
+        --data
+        local data = {}
+        data.type = "Started_game"
+        data.players = player_names
+        data.name = name
+        data.arguments = parse_args
+        data.server = server
+        game.write_file("mini_games/starting_game", game.table_to_json(data), false)
     else
-        if mini_game.options ~= 0 then
-            return "Wrong number of arguments"
+        local mini_game = Mini_games.mini_games[name]
+        if mini_game == nil then
+            return "This mini_game does not exsit"
         end
-    end
 
-    if started_game[1] == name then
-        return "This game is already running"
-    end
+        if parse_args then
+            if  mini_game.options ~= #parse_args then
+                return "Wrong number of arguments"
+            end
+        else
+            if mini_game.options ~= 0 then
+                return "Wrong number of arguments"
+            end
+        end
 
-    if mini_game.map == nil then
-        error("No map set")
-    end
+        if started_game[1] == name then
+            return "This game is already running"
+        end
```

Comment:
Should `started_game` not be moved into `vars`?

Reply:
nope vars is just for the node server and the whole game shood be able to run without them